internal/multistorage/folder.go

```diff
@@ -98,9 +98,13 @@ func EnsureSingleStorageIsUsed(folder storage.Folder) error {
 func changeDirectory(path string, storages ...cache.NamedFolder) []cache.NamedFolder {
 	newStorages := make([]cache.NamedFolder, len(storages))
 	for i := range storages {
+		folder := storages[i].Folder
```

Comment:
There is a bug in some `storage.Folder` implementations (at least the s3 one): they work bad when calling `GetSubFolder` with the empty path "". 

In such case, the underlying path "" becomes "/", and the subsequent subfolders are appended to this "/" at the beginning making the underlying folder's path look like "/root/aaa/bbb". This breaks file listing and maybe something else. The problem is that listing is performed by matching the common suffixes between file and folder. And if folder path starts with "/", no files are matched to this folder.

I encountered this problem multiple times already with different storage implementations, because in this `multistorage.Folder` implementation it's very common to call `GetSubFolder("")`. So now I decided just keep `folder` variable the same if `path` is empty.